File: it/structures/java/inner-default/test/Entry.java
package test;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonProperty;
import java.util.Optional;

public class Entry {
    @JsonProperty("a")
    final Optional<A> a;
    @JsonProperty("b")
    final Optional<A.B> b;

    @JsonCreator
    public Entry(
        @JsonProperty("a") Optional<A> a,
        @JsonProperty("b") Optional<A.B> b
    ) {
        this.a = a;
        this.b = b;
    }

    @JsonProperty("a")
    public Optional<A> getA() {
        return this.a;
    }

    @JsonProperty("b")
    public Optional<A.B> getB() {
        return this.b;
    }

    @Override
    public String toString() {
        final StringBuilder b = new StringBuilder();

        b.append("Entry(");
        b.append("a=");
        b.append(this.a.toString());
        b.append(", ");
        b.append("b=");
        b.append(this.b.toString());
        b.append(")");

        return b.toString();
    }

    @Override
    public int hashCode() {
        int result = 1;
        final StringBuilder b = new StringBuilder();
        result = result * 31 + this.a.hashCode();
        result = result * 31 + this.b.hashCode();
        return result;
    }

    @Override
    public boolean equals(final Object other_) {
        if (other_ == null) {
            return false;
        }

        if (!(other_ instanceof Entry)) {
            return false;
        }

        @SuppressWarnings("unchecked")
        final Entry o_ = (Entry)other_;

        if (!this.a.equals(o_.a)) {
            return false;
        }

        if (!this.b.equals(o_.b)) {
            return false;
        }

        return true;
    }

    public static class Builder {
        private Optional<A> a;
        private Optional<A.B> b;

        private Builder() {
            this.a = Optional.empty();
            this.b = Optional.empty();
        }

        public Entry build() {

            return new Entry(
                this.a,
                this.b
            );
        }

        public Builder a(final A a) {
            this.a = Optional.of(a);
            return this;
        }

        public Builder b(final A.B b) {
            this.b = Optional.of(b);
            return this;
        }
    }

    /**
     * Construct a new builder.
     */
    public static Builder builder() {
        return new Builder();
    }
}

File: it/structures/csharp/inner-default/Test/Entry.cs
using Newtonsoft.Json;
using System;
using System.Text;

namespace Test {
    [JsonObject(ItemNullValueHandling = NullValueHandling.Ignore)]
    public class Entry {
        [JsonProperty("a")]
        public A a {
            get;
        }

        [JsonProperty("b")]
        public A.B b {
            get;
        }

        [JsonConstructor]
        public Entry (
            [JsonProperty("a")] A a,
            [JsonProperty("b")] A.B b
        ) {
            this.a = a;
            this.b = b;
        }

        public override bool Equals(Object other) {
            Entry o = other as Entry;

            if (o == null) {
                return false;
            }

            if (this.a == null) {
                if (o.a != null) {
                    return false;
                }
            } else {
                if (!this.a.Equals(o.a)) {
                    return false;
                }
            }

            if (this.b == null) {
                if (o.b != null) {
                    return false;
                }
            } else {
                if (!this.b.Equals(o.b)) {
                    return false;
                }
            }

            return true;
        }

        public override int GetHashCode() {
            int result = 1;
            result = result * 31 + this.a.GetHashCode();
            result = result * 31 + this.b.GetHashCode();
            return result;
        }

        public override String ToString() {
            StringBuilder b = new StringBuilder();

            b.Append("Entry(");
            b.Append("a=");
            b.Append(this.a);
            b.Append(", ");
            b.Append("b=");
            b.Append(this.b);
            b.Append(")");

            return b.ToString();
        }
    }
}
